### DISCUSSION TITLE
Process data in a different, more concise format with survival observations over time

### DISCUSSION THREAD

**Discussion by ggilestro** [CamDavidsonPilon/lifelines#1473] [Q&A]
My data observation have the following format

| Drug | hour |  alive | 
| aba | 0 | 71  |
| aba | 5 | 70 |
| aba | 10 | 68 |
| aba | 15 | 65 |
| aba | 20 | 60 |

I have 16 drugs, 3 concentrations for each drug and about 70 individuals per condition. 
Changing the file format to the default input would produce 5k-10k lines.

Is there any way to either read the CSV in the format I have or convert it into something that cab be understood by kfm?

Thanks

**Comment by CamDavidsonPilon** [comment]
Hi @ggilestro 

So your csv has ~3k rows already (16 * 3 * 70)? Is there a problem with producing 5k-10k lines?

**Reply by ggilestro** [reply]
No problem, just a way to say that I cannot possibly do this manually.
Anyway, other people have pointed me to the tinyr package in R which has a python equivalent in tidypython or one could follow [these hints](https://stackoverflow.com/questions/61533786/is-there-a-pandas-equivalent-to-tidyrs-uncount). 
Those would allow me to do what I mentioned.

**Reply by CamDavidsonPilon** [reply]
ooo is "alive" a count? This might be also called a "flatmap". 

If you can make it look something like:

| Drug | hour | deaths |
| aba | 0 | 0 |
| aba | 5 | 1 |
| aba | 10 | 2 |
| aba | 15 | 3 |
| aba | 20 | 5 |
...

that is probably easier to use with lifelines. Then the "deaths" is a weight column for any analysis. There's no censoring?
